include/libdnf/base/goal.hpp

```diff
@@ -54,53 +54,285 @@ class Goal {
     void add_rpm_remove(const libdnf::rpm::Package & rpm_package);
+
+    /// Add remove job to the goal. In default it uses `clean_requirements_on_remove` according to Config,
+    /// but it can be overridden by setting.
+    ///
+    /// @param rpm_package      A package to be added to the remove job.
+    /// @param settings         A structure to override default goal settings. Only clean_requirements_on_remove is used.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_erase(HyGoal goal, DnfPackage *pkg)
+    // @replaces libdnf/hy-goal.h:function:hy_goal_erase_flags(HyGoal goal, DnfPackage *pkg, int flags)
+    // @replaces dnf:dnf/base.py:method:Base().package_remove(self, pkg)
     void add_rpm_remove(const libdnf::rpm::Package & rpm_package, const libdnf::GoalJobSettings & settings);
+
+    /// Add remove job to the goal. In default it uses `clean_requirements_on_remove` according to Config.
+    ///
+    /// @param package_set      A package_set to be added to the remove job.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_erase_flags(HyGoal goal, DnfPackage *pkg, int flags)
     void add_rpm_remove(const libdnf::rpm::PackageSet & package_set);
+
+    /// Add remove job to the goal. In default it uses `clean_requirements_on_remove` according to Config,
+    /// but it can be overridden by setting.
+    ///
+    /// @param package_set      A package_set to be added to the remove job.
+    /// @param settings         A structure to override default goal settings. Only `clean_requirements_on_remove` is used.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_erase_flags(HyGoal goal, DnfPackage *pkg, int flags)
     void add_rpm_remove(const libdnf::rpm::PackageSet & package_set, const libdnf::GoalJobSettings & settings);
+
+    /// Add upgrade job to the goal for future resolvement. In default it uses `clean_requirements_on_remove` with `false` value.
+    ///
+    /// @param spec      A string describing the Goal upgrade job.
+    // @replaces dnf:dnf/base.py:method:Base().upgrade(self, pkg_spec, reponame=None)
     void add_rpm_upgrade(const std::string & spec);
+
+    /// Add upgrade job to the goal for future resolvement. In default it uses `clean_requirements_on_remove` with `false` value,
+    /// but it can be overridden by setting.
+    ///
+    /// @param spec      A string describing the Goal upgrade job.
+    /// @param settings  A structure to override default goal settings. Values  `from_repo_ids` and `strict` are not used.
+    // @replaces dnf:dnf/base.py:method:Base().upgrade(self, pkg_spec, reponame=None)
     void add_rpm_upgrade(const std::string & spec, const libdnf::GoalJobSettings & settings);
+
+    /// Add upgrade job of all installed packages to the goal. In default it uses `clean_requirements_on_remove` with `false` value.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_upgrade_all(HyGoal goal)
+    // @replaces dnf:dnf/base.py:method:Base().upgrade_all(self, reponame=None)
     void add_rpm_upgrade();
```

Comment:
Somewhat unrelated to documentation, but should the method be called `add_rpm_upgrade_all()` to differentiate it from the others that always take some sort of specifier or package(s)?

Reply:
I am not sure whether it is a good idea. `add_rpm_upgrade` is limited by `Settings.from_repo` and `to_repo`. Then all can be confusing.